src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -267,11 +267,15 @@ public void DeleteSearchParameter(string url, bool calculateHash = true)
                 throw new ResourceNotFoundException(string.Format(Core.Resources.CustomSearchParameterNotfound, url));
             }
 
-            // for search parameters with a base resource type we need to delete the search parameter
-            // from all derived types as well, so we iterate across all resources
-            foreach (var resourceType in TypeLookup.Keys)
+            // find all derived resources from the list of base resources
+            var allResourceTypes = GetDerivedResourceTypes(searchParameterInfo.BaseResourceTypes);
+            foreach (var resourceType in allResourceTypes)
             {
                 TypeLookup[resourceType].TryRemove(searchParameterInfo.Code, out var removedParam);
+                if (removedParam.Url != searchParameterInfo.Url)
+                {
+                    _logger.LogError("Error, Search Param {RemovedParam} removed from Search Param Definitoni manager.  It does not match deleted Search Param {Url}", removedParam.Url, url);
+                }
```

Comment:
@feordin  do we need to change CalculateSearchParameterHash method as well to update hash for only resource types that are affected?

Reply:
@rajithaalurims CalculateSearchParameterHash can be called when any change occurs to the Search Params and it will reproducible calculate the hash value for each resource type.  If no changes have occurred for a specific resource type, it won't be affected.  So we shouldn't need any changes here.